src/mesh/BoundingBox.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+#include "logging/Logger.hpp"
+#include "mesh/Vertex.hpp"
+#include <vector>
+
+namespace precice {
+namespace mesh {
+
+/// Bounding box class for mesh and partions
+class BoundingBox {
+
+public:
+  /***
+  * 
+  * @brief Constructor.
+  * 
+  * @param[in] bounds Vector of minimum and maximum points in each dimension
+  * 
+  */
+  BoundingBox(std::vector<double> bounds);
+  
+  /***
+  * 
+  * @brief Constructor.
+  * 
+  * @param[in] dimension Dimension of the bounding box
+  * 
+  */
+  BoundingBox(int dimension);
+
+  /// Default constructor
+  BoundingBox();
+
+  /// Special function for tests to create custom bounds
+  void modifyForTest(int rank, std::string testName);
+
+  /// Copy Constructor
+  BoundingBox(const BoundingBox &bb);
+
+  /// Destructor
+  ~BoundingBox();
```

Comment:
I guess this is added by default. So, normally we don't make it explicit, @fsimonis correct?